internal/controller/promotion/render.go

```diff
@@ -97,6 +99,7 @@ func (r *renderer) apply(
 		Images:       images,
 		LocalInPath:  workingDir,
 		LocalOutPath: writeDir,
+		RepoCreds:    repoCreds,
```

Comment:
One of two key changes. Most everything else is just updates to function signatures to get these creds down to here.